docs/contributing/continuous-integration.rst

```diff
@@ -1,32 +1,54 @@
-.. _pulpcore: https://github.com/PulpQE/pulp-smash/tree/master/pulp_smash/tests/pulp3/pulpcore
-.. _file: https://github.com/PulpQE/pulp-smash/tree/master/pulp_smash/tests/pulp3/file
+.. _istqb: https://www.istqb.org/downloads/syllabi/foundation-level-syllabus.html
 .. _pulp-smash: https://github.com/PulpQE/pulp-smash/
-
-
 .. _continuous-integration:
 
 Continuous Integration
 ======================
 
-New code is highly encouraged to have basic unit tests that demonstrate its functionality. A Pull
-Request that has failing unit tests cannot be merged.
+Unit Tests
+----------
+
+New code is highly encouraged to have basic unit tests that demonstrate that
+units (function, method or class instance) are working correctly.
+A Pull Request that has failing unit tests cannot be merged.
+
+The unit tests for `pulpcore` are in `pulpcore/tests
+<https://github.com/pulp/pulp/tree/master/pulpcore/tests/unit>`_.
+
+The unit tests for `pulpcore-plugin` are in `pulpcore-plugins/tests
+<https://github.com/pulp/pulp/tree/master/plugin/tests/unit/>`_.
+
+Functional Tests
+----------------
 
-The unit tests for `pulpcore` are in `pulpcore/tests <https://github.com/pulp/pulp/tree/master/pulpcore/tests>`_.
+Functional tests verify a specific action or feature.
+In general functional tests tend to answer the question "As an user can I do this?"
 
-Integration tests for new code should be added to a separate project called pulp-smash_. A Pull
-Request that has failing pulp-smash tests cannot be merged.
+Functional tests for Pulp are written using pulp-smash_ . Pulp smash is a test
+framework written to ease the test of Pulp.
 
-The integration tests for the REST API live in the pulpcore_ folder and the Plugin API is tested
-with tests in the file_ folder.
+It is highly encouraged to new features to be tested adding a functional test
```

Comment:
It is highly encouraged to accompany new features with functional tests in `pulpcore/functional <https://github.com/pulp/pulp/tree/master/pulpcore/tests/functional>`_.